docs/content/TextInput.mdx

```diff
@@ -108,6 +108,58 @@ const WithIconAndLoadingIndicator = () => {
 render(<WithIconAndLoadingIndicator />)
 ```
 
+### With trailing action
+
+```jsx live
+<Box display="grid" sx={{gap: 3}}>
+  <FormControl>
+    <FormControl.Label>Icon action</FormControl.Label>
+    <TextInput
+      trailingAction={
+        <TextInput.Action
+          onClick={() => {
+            alert('clear input')
+          }}
+          icon={XIcon}
+          iconLabel="Clear input"
+          sx={{color: 'fg.subtle'}}
+        />
+      }
+    />
```

Comment:
I'm not super confident about this but what do you think about an API like this:

```
```
```suggestion
    <TextInput>
        <TextInput.TrailingAction
          onClick={() => {
            alert('clear input')
          }}
          icon={XIcon}
          iconLabel="Clear input"
          sx={{color: 'fg.subtle'}}
        />
    </TextInput>
```

Reply:
Oh yeah! Me and @mperrotti went through a bunch of options and this was one of them! 

What I like:
1. It makes TextInput feel like a compound component and then putting children inside it makes sense

What I don't like:
1. Doesn't match the prop for trailingVisual (we _could_ move that inside as well along with leadingVisual)
2. It's not intuitive to put contents inside a TextInput because input can't have children.

Reply:
@siddharthkp said exactly what I was going to 😅